src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py

```diff
@@ -47,12 +47,18 @@ def list_accounts(client, resource_group_name=None):
 
 @transfer_doc(AutoStorageBaseProperties)
 def create_account(client, resource_group_name, account_name, location,  # pylint:disable=too-many-arguments
-                   tags=None, storage_account=None):
+                   tags=None, storage_account=None, keyvault=None):
+    from azure.cli.core._profile import CLOUD
     properties = AutoStorageBaseProperties(storage_account_id=storage_account) \
         if storage_account else None
     parameters = BatchAccountCreateParameters(location=location,
                                               tags=tags,
                                               auto_storage=properties)
+    if keyvault:
+        vault = keyvault.split('Microsoft.KeyVault/vaults/')[-1]
+        url = 'https://{}{}/'.format(vault, CLOUD.suffixes.keyvault_dns)
```

Comment:
This is how the URL is constructed in the KeyVault command module - so it should be safe.